src/site/_data/resource.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Generates the site version, based on the layout itself plus CSS/JS.
+ *
+ * This is used in internal navigation: if we see a change in the version, we throw out the HTML
+ * and do a proper fetch.
```

Comment:
It might worth expanding on this a bit in the comment to explain why we do this.

My understanding is we're still precaching everything, but instead of trying to piece together a shell that we send back to the client, we just inject a build version into every page. If we ever get a content page that's running a different version, we throw our hands up and do a full refresh, this causes the new html to come down without the sw involved so the page can fetch the correct assets and install a new sw (with a potentially new precache if needed).

Something like that...? 😅

Reply:
And the alternative to doing this is we make everything stale and use a prompt if someone wants to see the latest greatest, right?

Reply:
I've updated this comment a bit.